src/components/jellyfinActions.ts

```diff
@@ -76,10 +78,8 @@ export function reportPlaybackStart(
 
     restartPingInterval(reportingParams);
 
-    return JellyfinApi.authAjax('Sessions/Playing', {
-        contentType: 'application/json',
-        data: JSON.stringify(reportingParams),
-        type: 'POST'
+    await getPlaystateApi(JellyfinApi.jellyfinApi).reportPlaybackStart({
```

Comment:
Why do these not return anything now?

Reply:
My understanding is that there's an implicit return, so there's no need to include one since the caller doesn't read anything from the `void` return type. I can add an explicit one if that's preferred?

I'm JS beginner level so I apologise for any basic mistakes - I'm coming from a Kotlin background where there's no need to return from a function with `Unit` (`void`-equivalent) type.

Reply:
It isn't necessary to explicitly return but it does seem odd to use async/await in this way when you could just return the sdk call.

Reply:
I believe these specific API calls have no response (204) so it would be void anyway

Reply:
[`reportPlaybackStart`](https://github.com/jellyfin/jellyfin-sdk-typescript/blob/v0.10.0/src/generated-client/api/playstate-api.ts#L1181) and [`reportPlaybackProgress`](https://github.com/jellyfin/jellyfin-sdk-typescript/blob/v0.10.0/src/generated-client/api/playstate-api.ts#L1169) both return `void`, but in the generated typings they return `Promise<import("axios").AxiosResponse<void, any>>`.

Reply:
This is essentially the same as `return await asyncFn()` vs `return asyncFn()` which has some slight performance implications: https://stackoverflow.com/questions/38708550/difference-between-return-await-promise-and-return-promise

In my opinion returning the call is also just generally more explicit about the intent of the function.

The typing is correct for the functions `Promise<import("axios").AxiosResponse<void, any>>`:
* `Promise`: all async functions implicitly return a Promise
* `AxiosResponse<void, any>`: this is the response of an Axios request that returns `void`

Reply:
I'm still missing something.

If I return the call (with or without async/await):
```typescript
    return await getPlaystateApi(JellyfinApi.jellyfinApi).reportPlaybackProgress({
        playbackProgressInfo: reportingParams
    });
```

Then the function return type has to be `Promise<AxiosResponse<void>>`, meaning the `AxiosReponse` has to be handled at the call site. Is that the right way to go? I'd have thought that `AxiosResponse` should be handled in the API module and pass the DTO back to the caller instead (similar to what I've done with `getPlaybackInfo` which returns `Promise<any>` but will be corrected later to return `Promise<PlaybackInfoResponse>`).

Reply:
Having metadata about the request could be useful in some cases which is probably why the generator returns AxiosResponse by default. Since you don't need it this is probably fine. I do still think the code is a bit ambiguous to read but not the end of the world.